packages/cli/internal/pkg/cli/context/info.go

```diff
@@ -18,6 +22,18 @@ func (i Summary) IsEmpty() bool {
 	return reflect.ValueOf(i).IsZero()
 }
 
+//IsServerProcessEngine Does the workflow engine run as a server process? A server process has a potential one to many
+// mapping with workflow runs.
+func (i *Summary) IsServerProcessEngine() bool {
+	return serverEngines[strings.ToLower(strings.TrimSpace(i.Engines[0].Engine))]
+}
+
+//IsHeadProcessEngine Does the workflow engine run as a head process? A head process has one to one
+// mapping with workflow runs. Processes are not reused.
+func (i *Summary) IsHeadProcessEngine() bool {
+	return headEngines[strings.ToLower(strings.TrimSpace(i.Engines[0].Engine))]
```

Comment:
If we expect Engine name to be in a wrong format (padding spaces, mixed case), I think it will be better to normalize it on ingestion, so we can avoid repeating this boilerplate code in all places. There is a potential risk to create inconsistent behavior and inflate number of unit tests. Instead of testing every method that touches Engine for trim and lowerCase, we can test one.